pkg/reconciler/v1alpha1/route/route.go

```diff
@@ -129,11 +130,29 @@ func NewController(
 	})
 
 	c.Logger.Info("Setting up ConfigMap receivers")
-	c.configStore = config.NewStore(c.Logger.Named("config-store"))
+	c.configStore = config.NewStore(c.Logger.Named("config-store"), c.enqueueRoutes(impl))
 	c.configStore.WatchConfigs(opt.ConfigMapWatcher)
 	return impl
 }
 
+func (c *Reconciler) enqueueRoutes(impl *controller.Impl) func(string, interface{}) {
+	return func(_ string, value interface{}) {
+		_, ok := value.(*config.Domain)
+		if !ok {
+			c.Logger.Debugf("not a *config.Domain: '%#v'", value)
+			return
```

Comment:
It wasn't immediately obvious to me from this log statement that this is not an error path, but a **filter**.

I think that I'd like to see us decompose and encapsulate this a bit more so that the `NewStore` call (above) can look like:
```golang
// Triggers a global resync of the objects in the SharedInformer's Store anytime we store a new config:
c.configStore = config.NewStore(c.Logger.Named("config-store"),
  config.GlobalResync(routeInformer))

// Filter the GlobalResync to just changes in the config.Domain ConfigMap (last so it can be variadic):
c.configStore = config.NewStore(c.Logger.Named("config-store"),
  config.TypeFilter(impl.GlobalResync(routeInformer), &config.Domain{}, ...))
```

I believe we can do this in a general way by having ([see also](https://godoc.org/k8s.io/client-go/tools/cache#SharedInformer)):
```go
func (c *Impl) GlobalResync(si cache.SharedInformer) yourFunc {
  return func(string, interface{}) {
    store := si.GetStore()
    // Consider also ListKeys, but I'm not sure its keys are guaranteed to have the same sense as ours.
    for _, obj := range store.List() {
      c.Enqueue(obj)
    }
  }
}
```

As proposed above, we'd need to get some code into `knative/pkg`.  I'd be fine with staging this in `knative/serving`, with loose agreement about how we move it and then doing the `knative/pkg` reshuffling.  This is partly because I suspect we may want to generalize some of the config store stuff into `knative/pkg` as well.